lib/rules/no-useless-escape.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * @fileoverview Look for useless escapes in strings and regexes
+ * @author Onur Temizkan
+ * @copyright 2016 Onur Temizkan. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+var VALID_STRING_ESCAPES = {
```

Comment:
I thought that map lookup would be a faster choice than an array search. Not sure how much though.